nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/DetectDuplicateRecord.java

```diff
@@ -0,0 +1,620 @@
+            .required(false)
+            .addValidator(StandardValidators.createAttributeExpressionLanguageValidator(ResultType.STRING, true))
+            .defaultValue("${hash.value}")
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .build();
+
+    static final PropertyDescriptor AGE_OFF_DURATION = new PropertyDescriptor.Builder()
+            .name("Age Off Duration")
+            .description("Time interval to age off cached filters. When the cache expires, the entire filter and its values " +
+                    "are destroyed. Leaving this value empty will cause the cached entries to never expire.")
+            .required(false)
+            .addValidator(StandardValidators.TIME_PERIOD_VALIDATOR)
+            .build();
+
+    static final PropertyDescriptor RECORD_HASHING_ALGORITHM = new PropertyDescriptor.Builder()
+            .name("Record Hashing Algorithm")
+            .description("The algorithm used to hash the combined result of RecordPath values in the cache.")
+            .allowableValues(
+                    MD5_ALGORITHM_VALUE,
+                    SHA1_ALGORITHM_VALUE,
+                    SHA256_ALGORITHM_VALUE,
+                    SHA512_ALGORITHM_VALUE
+            )
+            .defaultValue(SHA1_ALGORITHM_VALUE.getValue())
+            .expressionLanguageSupported(ExpressionLanguageScope.NONE)
+            .required(true)
+            .build();
+
+    static final PropertyDescriptor FILTER_TYPE = new PropertyDescriptor.Builder()
+            .name("Filter Type")
+            .description("The filter used to determine whether a record has been seen before based on the matching RecordPath criteria.")
+            .allowableValues(
+                    HASH_SET_VALUE,
+                    BLOOM_FILTER_VALUE
+            )
+            .defaultValue(HASH_SET_VALUE.getValue())
+            .required(true)
+            .build();
+
+    static final PropertyDescriptor FILTER_CAPACITY_HINT = new PropertyDescriptor.Builder()
+            .name("Filter Capacity Hint")
+            .description("An estimation of the total number of unique records to be processed. " +
+                    "The more accurate this number is leads to less resizing operations on a HashSet filter and fewer false negatives on a BloomFilter. " +
+                    "Using the CountText processor block first and then using the ${text.line.count} attribute may be ideal for more dynamic estimates.")
+            .defaultValue("25000")
+            .expressionLanguageSupported(ExpressionLanguageScope.NONE)
+            .addValidator(StandardValidators.INTEGER_VALIDATOR)
+            .required(true)
+            .build();
+
+    static final PropertyDescriptor BLOOM_FILTER_FPP = new PropertyDescriptor.Builder()
+            .name("BloomFilter Probability")
+            .description("The desired false positive probability when using the BloomFilter filter type. " +
```

Comment:
Does it make sense to instead call this 'Bloom Filter Certainty' so the user can enter it in precisely the terms the description states rather than the negative of it?  Also, can you describe the tradeoff the user is making terms of time/space so they know the cost of different values?

Reply:
Also, on the documentation side you need to make it explicit to users that there is a possibility that there can be false positives if they choose this trade off.